### PR TITLE
Made some modifications to reduce the amount of code

### PR DESCRIPTION
**Pull request by margorczynski**
I've tried to reduce the amount of code and increase readability by for example removing the for-comprehension that basically only "unpacked" the Xor from the DBIO, replaced with a map.
Also I've tried to show how Cats supports transforming Option-s into Xor-s using Xor.fromOption.
I've changed the validatePassword to return a Xor just to keep it consistent with the rest (everything returns a Xor, it is transformed into Validated, glued together and again transformed into a Xor).
What do You think @jczuchnowski ?

### CODE REVIEW

File: src/main/scala/io/scalac/wtf/domain/User.scala [hunk 1]
```diff
@@ -24,45 +25,40 @@ case class NewUser(email: String, password: String)
...
+    validation.map(_.toXor)
   }
 
   private def validateEmail(email: String)(implicit ec: ExecutionContext): XorT[DBIO, ValidationError, String] = {
-    val emailPatternValidation = (emailPattern findFirstIn email) match {
-      case Some(_) => Xor.right(email)
-      case None => Xor.left(WrongEmailPattern)
-    }
+
+    val emailPatternValidation = Xor.fromOption(emailPattern findFirstIn email, WrongEmailPattern)
```
Comment (jczuchnowski) [inline src/main/scala/io/scalac/wtf/domain/User.scala:45]: Awesome. Didn't know about this one.

File: src/main/scala/io/scalac/wtf/domain/User.scala [hunk 2]
```diff
@@ -24,45 +25,40 @@ case class NewUser(email: String, password: String)
...
-        case None => Xor.right(email)
+        case Some(_) => left(UserAlreadyExists)
+        case None    => right(email)
     }
 
     for {
       e1 <- XorT[DBIO, ValidationError, String](DBIO.successful(emailPatternValidation))
       e2 <- XorT[DBIO, ValidationError, String](emailUniqueValidation)
-    } yield (e1 |+| e2)
+    } yield e1 |+| e2
```
Comment (jczuchnowski) [inline src/main/scala/io/scalac/wtf/domain/User.scala:55]: :+1:

File: src/main/scala/io/scalac/wtf/domain/User.scala [hunk 3]
```diff
@@ -24,45 +25,40 @@ case class NewUser(email: String, password: String)
...
   def validateUser(email: String, password: String)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], User]] = {
-    val validatedUser = for {
-      emailV <- validateEmail(email).toValidated
-      emailNel = emailV.toValidatedNel
-      passwordNel = validatePassword(password).toValidatedNel
-    } yield (emailNel |@| passwordNel) map { 
-      case (email, password) => User(email = email, password = password)
+
+    val validation = validateEmail(email).toValidated map { emailV =>
+      (emailV.toValidatedNel |@| validatePassword(password).toValidatedNel) map (User.apply(None, _, _))
```
Comment (jczuchnowski) [inline src/main/scala/io/scalac/wtf/domain/User.scala:37]: I can see your point, but I think these two versions are a matter of personal preference.
My version has more lines but the lines are more straightforward. And I do like for-comprehensions :)

You version is shorter no doubt about it.

I wonder what @pjazdzewski1990 thinks about it.

Comment (pjazdzewski1990) [inline src/main/scala/io/scalac/wtf/domain/User.scala:37]: +1 for the shorter one

**Comment by pjazdzewski1990** [comment]
Looks good

**Pull request merged**